lib/autoupdate/tools/updater.go

```diff
@@ -418,9 +420,11 @@ func (u *Updater) Exec(ctx context.Context, toolsVersion string, args []string)
 	}
 
 	env := filterEnvs(os.Environ(), []string{
+		teleportToolsPathReExecEnv,
 		teleportToolsVersionReExecEnv,
 		teleportToolsDirsEnv,
 	})
+	env = append(env, fmt.Sprintf("%s=%s", teleportToolsPathReExecEnv, executablePath))
```

Comment:
nit: if we're just concatenating strings, going through `fmt` is a bit slower than using concatenation (in a single expression)
```suggestion
	env = append(env, teleportToolsPathReExecEnv + "=" + executablePath)
```